google/cloud/storage/internal/object_streambuf.cc

```diff
@@ -68,9 +68,10 @@ StatusOr<ObjectReadStreambuf::int_type> ObjectReadStreambuf::Peek() {
   // assert(n <= current_ios_buffer_.size())
   current_ios_buffer_.resize(read_result->bytes_received);
 
-  for (auto const& kv : read_result->response.headers) {
+  auto hint = headers_.end();
+  for (auto& kv : read_result->response.headers) {
     hash_validator_->ProcessHeader(kv.first, kv.second);
-    headers_.emplace(kv.first, kv.second);
+    hint = headers_.emplace_hint(hint, kv.first, std::move(kv.second));
```

Comment:
Given that the `kv.first` values are in order, it seems like a better hint for the next insertion would be the successor of the last insertion.

Reply:
Done. Keeping my fingers crossed ... hopefully all the compilers we use support `std::next()`.